packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx

```diff
@@ -0,0 +1,134 @@
+
+const AddSoilAmendmentProducts = ({
+  products,
+  isReadOnly,
+  ...props
+}: AddSoilAmendmentProductsProps) => {
+  const [invalidProducts, setInvalidProducts] = useState<string[]>([]);
+
+  const { t } = useTranslation();
+  const { control, setValue, watch } = useFormContext();
+  const { fields, append, remove } = useFieldArray({
+    name: FIELD_NAME,
+    control,
+  });
+
+  const { expandedIds, toggleExpanded, expand, unExpand, resetExpanded } = useExpandable({
+    isSingleExpandable: true,
+  });
+
+  const productsForTask = watch(FIELD_NAME);
+
+  const getAvailableProductOptions = (selectedProductId: ProductId): Product[] => {
+    // Returns the products that have not been selected by any other select
+    const otherSelectedProductIds = productsForTask
+      .filter(({ product_id }: ProductFields): boolean => product_id !== selectedProductId)
+      .map(({ product_id }: ProductFields): ProductId => product_id);
+
+    return products.filter(({ product_id }) => !otherSelectedProductIds.includes(product_id));
+  };
+
+  const getInvalidProductsUpdater =
+    (fieldId: string) =>
+    (isValid: boolean): void => {
+      setInvalidProducts((prevState) => {
+        if (isValid && prevState.includes(fieldId)) {
+          return prevState.filter((id) => id !== fieldId);
+        }
+        if (!isValid && !prevState.includes(fieldId)) {
+          return [...prevState, fieldId];
+        }
+        return prevState;
+      });
+    };
+
+  const onRemove = (index: number, fieldId: string): void => {
+    if (invalidProducts.includes(fieldId)) {
+      setInvalidProducts(invalidProducts.filter((id) => id !== fieldId));
+    }
+    remove(index);
+  };
+
+  const onAddAnotherProduct = (): void => {
+    resetExpanded();
+    append(defaultValues);
+  };
+
+  return (
+    <>
+      <div className={styles.products}>
+        {fields.map((field, index) => {
+          const namePrefix = `${FIELD_NAME}.${index}`;
+          const productId = productsForTask[index]?.product_id;
+
+          return (
+            <ProductCard
+              {...props}
+              key={field.id}
+              isReadOnly={isReadOnly}
+              onRemove={index ? () => onRemove(index, field.id) : undefined}
+              namePrefix={namePrefix}
+              products={getAvailableProductOptions(productId)}
+              isExpanded={expandedIds.includes(field.id)}
+              toggleExpanded={() => toggleExpanded(field.id)}
+              unExpand={() => unExpand(field.id)}
+              expand={() => expand(field.id)}
+              productId={productId}
+              setProductId={(id: number | string | undefined) => {
+                setValue(`${namePrefix}.product_id`, id);
+              }}
+              setFieldValidity={getInvalidProductsUpdater(field.id)}
+            />
+          );
+        })}
+      </div>
+      {!isReadOnly && (
+        <TextButton
+          disabled={!!invalidProducts.length}
```

Comment:
Is it correct that a farm not pursuing certification can add an infinite number of empty product cards since the `invalidProducts` array is empty?

<img width="945" alt="Screenshot 2024-06-14 at 10 05 35 AM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/3cb79407-24b8-403f-8777-2a6d15366708">

Reply:
No, it's a bug! The button shouldn't be enabled if a product hasn't been selected!
Thank you :)